pandas/core/internals.py

```diff
@@ -3453,7 +3482,9 @@ def apply(self, f, axes=None, filter=None, do_integrity_check=False,
 
         aligned_args = dict((k, kwargs[k])
                             for k in align_keys
-                            if hasattr(kwargs[k], 'values'))
+                            if hasattr(kwargs[k], 'values')
+                            # eww
+                            and not isinstance(kwargs[k], ExtensionArray))
```

Comment:
So, this was causing the failures with `series[mask] = value`. Basically, we relied on our NumPy arrays not having / adding a `.values` attribute.

I'm thinking about splitting `__setitem__`-related changes off into their own PR since this is growing out of control.

Reply:
Another option is to keep the example extension array with dicts unsettable (or skip the tests for them), and keep with "settting with sequences" for later. As I suppose the more simple setting case where the scalar type is not a sequence is less of a problem?